torch/utils/checkpoint.py

```diff
@@ -14,10 +15,16 @@ def detach_variable(inputs):
             "Only tuple of tensors is supported. Got Unsupported input type: ", type(inputs).__name__)
 
 
+def check_backward_validity(inputs):
+    if not any(inp.requires_grad for inp in inputs):
+        warnings.warn("None of the inputs have requires_grad=True. Gradients will be None")
+
+
 class CheckpointFunction(torch.autograd.Function):
 
     @staticmethod
     def forward(ctx, run_function, *args):
+        check_backward_validity(*args)
```

Comment:
Wait this is wrong! You should pass in at least `args` instead of `*args`. It should be a Python error if `len(args) != 1`. 

Also, I'm curious. Why not `**kwargs` support?

Reply:
@SsnL , the *args here are the inputs to the run_function. the inputs are tensors. why should len(args)==1 ? a model can have many input tensors. 

why should **kwargs be supported?

Reply:
check_backward_validity(*args) doesn't make sense. The only reason everything worked is that there are no unit tests for variable number of args in unit tests (all unit tests have one input argument).
Even in one-argument case, `check_backward_validity` didn't have an error because Tensor is iterable.

I've fixed it, `check_backward_validity(args)` is correct, as `inputs` will then be a tuple.

Priya, can you send a PR to add unit tests for multiple input case (no hurry)

Reply:
thanks for fixing. I understand now what @SsnL meant, earlier I thought he was referring to input to forward(). I'll send out the PR tomorrow for multiple input cases. I tested this case so far with wlm, vnet models.

Reply:
@SsnL the main thing preventing us from using kwargs is that Functions don’t support them, because we need to have a consistent ordering of inputs.

Reply:
@apaszke I see. Thanks for explanation. If only we can all use [python 3.6](https://www.python.org/dev/peps/pep-0468/).